app/Foliage/CmdBuild.hs

```diff
@@ -290,8 +297,48 @@ prepareIndexPkgMetadata expiryTime PreparedPackageVersion {pkgId, sdistPath} = d
         targetsDelegations = Nothing
       }
 
+prepareIndexPkgExtraEntries :: UTCTime -> [PreparedPackageVersion] -> [Tar.Entry]
+prepareIndexPkgExtraEntries timestamp packageVersions = map buildTarEntry $ Map.toList $ getPreferredVersions packageVersions
+  where
+    buildTarEntry (pkgName, pkgVersions) = mkTarEntry' content path timestamp
+      --                                                            ^^^^^^^^^ TODO: this is for now set to `currentTime` ...
+      --                                                            ... should it be the timestamp of the latest version of the package?
```

Comment:
drive-by comment: we'll need one for every time something gets deprecated, as duncan said on slack. We need to put them into the tarball in the right order... Otherwise index-state won't work